### DISCUSSION TITLE
Nakamoto Testing - Core Requirements of a Mainnet Representative Testing

### DISCUSSION THREAD

**Discussion by AshtonStephens** [stacks-network/stacks-core#4049] [Ideas]
This ticket discusses and tracks the core requirements and ideas that the Stacks Testing needs to meet such that the automatically deployed infrastructure mimics mainnet as closely as possible.

**Relevant Draft SIP:** SIP-021 https://github.com/stacksgov/sips/pull/155
**Somewhat Relevant Discussion:** https://github.com/stacksgov/sips/discussions/161
**Testing Standards Document:** https://github.com/stacks-network/stacks/pull/515 (Should replace with the approved location)

Big questions:
- Do we need a full existing mainnet chain state for each test node or start fro a new genesis block
    - If a new genesis sync then we'll need at least one miner
- What types of tests are we running and what is the expected behavior?
- What metrics and logs are needed by open source developers?
- What outputs from the test need to be recorded / what events should be exposed

**Comment by AshtonStephens** [comment]
Just started this after a discussion with @wileyj

**Reply by AshtonStephens** [reply]
We'll need to discuss with @xoloki and @jferrant about signer tests and requirements.

**Reply by wileyj** [reply]
I've gathered some additional/changed details over the past few days. I expect I'll have a more complete picture later this week

**Comment by wileyj** [comment]
Suite of tests for 2.4 release: https://github.com/stacks-network/stacks-test-scripts-2.1
